src/App.tsx

```diff
@@ -5,7 +5,13 @@ import { FindMovie } from './components/FindMovie';
 import { Movie } from './types/Movie';
 
 export const App = () => {
-  const [movies] = useState<Movie[]>([]);
+  const [movies, setMovies] = useState<Movie[]>([]);
+
+  const handleAddMovie = (newMovie: Movie) => {
+    if (!movies.find(movie => movie.imdbId === newMovie.imdbId)) {
+      setMovies([...movies, newMovie]);
```

Comment:
If the new state is computed use the previous state, you can pass a function to `setMovies`.